dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceCache.java

```diff
@@ -0,0 +1,220 @@
+        String group = primaryGroup;
+
+        Resource resource = null;
+        String key = group + cleanedResourceKey;
+        try {
+            resource = (Resource) cache.get(key, group);
+        } catch (DotCacheException e) {
+            Logger.debug(this, "Cache Entry not found", e);
+        }
+        return resource != null ? resource : null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.velocity.runtime.resource.ResourceCache#initialize(org.apache.velocity.runtime.
+     * RuntimeServices)
+     */
+    public void initialize(RuntimeServices rs) {
+        cache = CacheLocator.getCacheAdministrator();
+    }
+
+    public void addMiss(Object resourceKey) {
+        Logger.info(this.getClass(), "velocityMiss:" + resourceKey);
+    }
+
+    public boolean isMiss(Object resourceKey) {
+        return false;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#put(java.lang.Object,
+     * org.apache.velocity.runtime.resource.Resource)
+     */
+    public Resource put(Object resourceKey, Resource resource) {
+        if (resource != null && ignoreGlobalVM.contains(resource.getName())) {
+            return resource;
+        }
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+
+        String key = group + cleanedResourceKey;
+        // Add the key to the cache
+        cache.put(key, resource, group);
+
+        return resource;
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#remove(java.lang.Object)
+     */
+    public Resource remove(Object resourceKey) {
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+        String key = group + cleanedResourceKey;
+
+        try {
+            cache.remove(key, group);
+        } catch (Exception e) {
+            Logger.debug(this, e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public void clearCache() {
+        for (String group : groupNames) {
+            cache.flushGroup(group);
+        }
+
+    }
+
+    @Deprecated
+    public void clearMenuCache() {
+
+    }
+
+    public String[] getGroups() {
+        return groupNames;
+    }
+
+    public String getPrimaryGroup() {
+        return primaryGroup;
+    }
+
+    private String cleanKey(String key) {
+        if (key.startsWith(ResourceManager.RESOURCE_TEMPLATE + ""))
+            key = key.substring((ResourceManager.RESOURCE_TEMPLATE + "").length());
+
+        if (key.startsWith(File.separatorChar + "")) {
+            key = key.substring(1);
+        }
+        if (key.startsWith("/")) {
+            key = key.substring(1);
+        }
+        key = StringUtil.replace(key, '\\', '/');
+        return key;
+    }
+
+    @Deprecated
+    public String getMenuGroup() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "getPrimaryGroup" on line 186. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)